app/Models/CustomerCart.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\Relations\BelongsToMany;
+use Illuminate\Support\Facades\Auth;
+
+class CustomerCart extends Model
+{
+    use HasFactory;
+    
+    protected $fillable = [
+        'product_id',
+        'qty'
+    ];
+
+    public function user():BelongsTo{
+        return $this->belongsTo(User::class);
+    }
+
+    // public function product():BelongsToMany{
+    //     return $this->belongsToMany(Product::class);
+    // }
+
+
+    public function scopeCartOwner($query){
+        $query->where('user_id',Auth::user()->id)->join('products',function($query){
```

Comment:
We'd suggest to have the user id be passed as a parameter instead. This scope function assumes that we have an authenticated user which limits its usability. Testing in tinker for example would be impossible because there wont be an `Auth::user` then.

On that note, it's typically best practice to not write functions that are self-contained. Let's keep a mindset of making our code readily available for wide use, and easily scalable for when we have projects that can expand its scope.